libraries/androidutils/src/main/kotlin/io/element/android/libraries/androidutils/system/SystemUtils.kt

```diff
@@ -47,6 +48,19 @@ fun Context.getApplicationLabel(packageName: String): String {
     }
 }
 
+/**
+ * Retrieve the versionCode from the Manifest.
+ * The value is more accurate than BuildConfig.VERSION_CODE, as it is correct according to the
+ * computation in the `androidComponents` block of the app build.gradle.kts file.
+ * In other words, the last digit (for the architecture) will be set, whereas BuildConfig.VERSION_CODE
+ * last digit will always be 0.
+ */
+fun Context.getVersionCodeFromManifest(): Long {
+    return PackageInfoCompat.getLongVersionCode(
```

Comment:
Are we sure it does work for every OS and with R8?

Reply:
I do not think R8 will remove this from the manifest, since it's used by the system to manage application upgrade.